components/webcompat/content/browser/webcompat_exceptions_service.h

```diff
@@ -39,30 +42,24 @@ class WebcompatExceptionsService
                         const std::string& manifest) override;
 
   ~WebcompatExceptionsService() override;
-  void OnJsonFileDataReady(const std::string& contents);
   static WebcompatExceptionsService* CreateInstance(
       brave_component_updater::LocalDataFilesService* local_data_files_service);
   static WebcompatExceptionsService* GetInstance();
+  // Callable from any thread; needed for functions like
+  // HostContentSettingsMap::GetContentSetting(...)
   const std::vector<ContentSettingsPattern> GetPatterns(
       ContentSettingsType webcompat_type);
-  bool AddRuleForTesting(const ContentSettingsPattern& pattern,
-                         const std::string& exception_string);
+  void SetRules(const PatternsByWebcompatTypeMap& patterns_by_webcompat_type);
```

Comment:
do you want to let anyone to be able to set rules at any point? Or is it public only for tests? If it's for tests, then make it public only for tests.

Reply:
OK, I added `SetRulesForTesting` and changed `SetRules` to private.